effekt/shared/src/main/scala/effekt/RecursiveDescent.scala

```diff
@@ -1105,6 +1106,18 @@ class RecursiveDescent(positions: Positions, tokens: Seq[Token], source: Source)
     case _ if isHole         => hole()
     case _ if isTupleOrGroup => tupleOrGroup()
     case _ if isListLiteral  => listLiteral()
+    case `${` =>
+      nonterminal:
+        val startPosition = position
+        manyUntil(next(), `}$`)
+        val endPosition = position
+        if peek(`}$`) then consume(`}$`) else if peek(`}`) then consume(`}`)
+
+        val msg = s"Unexpected splice: string interpolation $${ ... } is only allowed in strings!"
+        softFail(msg, startPosition, endPosition)
+
+        UnitLit(span().synthesized) // XXX(jiribenes, 2025-07-01): HACK!
```

Comment:
We should really add `Error` trees for this very purpose instead of misusing `UnitLit`